docs/usage.md

```diff
@@ -37,28 +40,75 @@ $ ./bin/ipc-agent subnet send-value --subnet <subnet-id> --to <to-addr> <value>
 $ ./bin/ipc-agent subnet send-value --subnet /root/t01002 --to t1xbevqterae2tanmh2kaqksnoacflrv6w2dflq4i 10
 ```
 
-## Leaving a subnet
+## Sending funds between subnets
 
-To leave a subnet, the following agent command can be used:
+At the moment, the IPC agent only expose commands to perform the basic IPC interoperability primitives for cross-net communication, which is the exchange of FIL (the native token for IPC) between the same address of a subnet. Mainly:
+- `fund`, which sends FIL from one public key address, to the same public key address in the child.
+- `release` that moves FIL from one account in a child subnet to its counter-part in the parent.
+
+Complex behavior can be implemented using these primitives: sending value to a user in another subnet can be implemented a set of `release/fund` and `sendValue` operations. Calling  smart contract from one subnet to another works by providing funds to one account in the destination subnet, and then calling the contract. The agent doesn't currently include abstractions for this complex operations, but it will in the future. That being said, users can still leverage the agent's API to easily compose the basic primitives into complex functionality.
+
+>💡 All cross-net operations need to pay an additional cross-msg fee (apart from the gas cost of the message). This is reason why even if you sent `X FIL` you may see `X - fee FIL` arriving to you account at destination. This fee is used to reward subnet validators for their work committing the checkpoint that carries the message.
+
+### Fund
+Funding a subnet can be performed by using the following command:
 ```bash
-$ ./bin/ipc-agent subnet leave --subnet <subnet-id>
+$ ./bin/ipc-agent cross-msg fund --subnet=<subnet-id> <amount>
 
 # Example execution
-$ ./bin/ipc-agent subnet leave --subnet /root/t01002
+$ ./bin/ipc-agent cross-msg fund --subnet=/root/t01002 100
+
 ```
-Leaving a subnet will release the collateral for the validator and remove all the validation rights from its account. This means that if you have a validator running in that subnet, its validation process will immediately terminate.
+This command includes the cross-net message into the next top-down checkpoint after the current epoch. Once the top-down checkpoint is committed, you should see the funds in your account of the child subnet.
+
+>💡 Top-down checkpoints are not used to anchor the security of the parent into the child (as is the case for bottom-up checkpoints). They just include information of the top-down messages that need to be executed in the child subnet, and are a way for validators in the subnet to reach consensus on the finality on their parent.
+
+### Release
+In order to release funds from a subnet, your account must hold enough funds inside it. Releasing funds to the parent subnet can be permformed with the following comand:
```

Comment:
So the command works as expected but if you try to release in excess of the supply, you get:
> [2023-04-19T00:02:18Z ERROR ipc_agent::manager::bottomup] cannot submit checkpoint due to json_rpc error: {"code":1,"message":"GasEstimateMessageGas error: message execution failed: exit 20, reason: message failed with backtrace:\n00: f01002 (method 1487690799) -- send to t064 method 1468071864 aborted with code 20 (20)\n01: f064 (method 1468071864) -- error releasing circulating supply: wtf! we can't release funds below circ, supply. something went really wrong (20)\n (RetCode=20)"}

This might be a bit too alarming -- it's not that something went really wrong, just a user made a wrong input.

Reply:
Wait, I might have misread it. In fact, per https://github.com/consensus-shipyard/lotus/pull/161, the subnet account is initialised with 5000 FIL at genesis but this isn't counted towards the circulating supply because it didn't go through `fund`. So then we end up with the `release` returning success (fund check clears) but the message then failing validation on account of the circulating supply. Not great 🤔

Reply:
I relaxed a bit the tone of that error messages (just in case someone sees it in the future), and since the fix of that PR, it should be hard for validators to encounter it as we are removing all genesis funds (except for the faucet).